routes/userApi.js

```diff
@@ -0,0 +1,12 @@
+import Router from 'express';
+import { createUser, loginUser } from '../controllers/userApiLogic.js';
+import validateSignUp from '../middleware/validations/validate-signup.js';
+import validateLogin from '../middleware/validations/validate-login.js';
+import adminAuth from '../middleware/admin-auth.js';
+
+const router = Router();
+
+router.route('/create-user').post(adminAuth, validateSignUp, createUser);
```

Comment:
on this route add the jwt validation middleware for authenticated route. do not use` req.body.isAdmin` inside adminAuth. now imagine a malicious user sends that to the api as part of a payload, the user can set isAdmin as part of the body to perform any activity. that is why you have to add the jwt middleware, that decodes the token and then appends the user object to the req which you can then access inside the admin user middleware

Reply:
but this route(register route) is not a protected route. a token is only generated after a user registers, then the token is decoded via the jwt validation middleware so we can know which user is assessing a particular route.  so i still dont understand how i can add the jwt middleware to this route sha, since we dont have a token yet. 
also, is it ONLY an admin that can register employees? or employees can  register themselves? if the latter is true, den i think there is no need having the admin middleware yet. it will come in later probably when an admin is needed to do a particular function like delete an employees comment. i just need some clarity on this sha